slidedeck/src/slides/02-react-basics/00-react-basics.pug

```diff
@@ -91,143 +91,104 @@ section
 
     class Hello extends React.Component {
       render() {
-        return <h1>Hello {this.props.name}!</h1>;
+        return <h1>Hello World!</h1>;
       }
     }
 
-    const container = document.getElementById('root');
-    ReactDOM.render(<Hello name="Tim" />, container);
-
-section
-  h2 Component Specifications
-
-section
-  p Component Features:
-  ul
-    li methods
-      ul
-        li <span class="component red">render()</span>
-        li lifecycle methods
-    li properties
-      ul
-        li <span class="component red">props</span>, <span class="component red">defaultProps</span>, & <span class="component red">propTypes</span>
-        li <span class="component red">state</span> & <span class="component red">setState()</span>
-
 section
   h2.lower render()
   ul
-    li Required
     li Returns child element(s) to display (which can be other components)
     li Or returns null or false to render nothing
-
-section
-  h2.lower render()
-  ul
-    li Does not mutate component state
     li Should return the same result each time it is invoked (rendering consistency!)
     li Should not read from or write to the DOM
 
 section
+  h2.lower render()
   script(type="text/plain", class="language-jsx").
     render() {
       return (
         <div>
-          <p>Can use props and state here</p>
-          <small>(but can't change them)</small>
+          <p>You can render HTML elements</p>
+          <OrCustomComponents/>
+          <p>But remember, this runs over and over</p>
+          <div>{callAPI("So don't do this")}</div>
         </div>
       );
     }
 
 section
-  h2 Component Lifecycle
+  h2.lower props
   ul
-    li Mounting
-    li Updating
-    li Unmounting
-
-section
-  h2 Mounting
-  p These methods are called when an instance of a component is being created and inserted into the DOM:
-  ol
-    li <span class="component red">constructor()</span>
-    li <span class="component red">render()</span>
-    li <span class="component red">componentDidMount()</span>
+    li an object containing the parameters that were defined by the caller of the component
+    li should never be modified by consuming component
 
 section
-  h2 Updating:
-  p An update can be caused by changes to props or state.
-  ol
-    li <span class="component red">shouldComponentUpdate()</span>
-    li <span class="component red">render()</span>
-    li <span class="component red">componentDidUpdate()</span>
+  h2.lower props
+  script(type="text/plain", class="language-jsx").
+    class MyParentComponent extends React.Component {
+      render() (
+        <MyNameComponent firstName="John" lastName="Smith"/> 
+      )
+    }
 
-section
-  h2 Unmounting:
-  p This method is called when an instance of a component is being destroyed and removed from the DOM:
-  ol
-    li <span class="component red">componentWillUnmount()</span>
+  script(type="text/plain", class="language-jsx").
+    class MyNameComponent extends React.Component {
+      render() (
```

Comment:
Maybe I'm not up to date on the latest ES hotness, but this syntax looks weird

Reply:
Hah, whoops